### PR TITLE
Adding template code to create a VPC module to import into an existing project

### PR DESCRIPTION
**Pull request by sylviamclaughlin**
# Summary | Résumé

A PR for a Backstage template to add a VPC module to an existing project.

### CODE REVIEW

File: templates/vpc/content/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,43 @@
+output "vpc_id" {
```
Comment (patheard) [inline templates/vpc/content/outputs.tf:1]: Minor nit: could you run `terraform fmt` on the `templates/vpc` folder

File: templates/vpc/content/vpc.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#
+# VPC
+#
+
+module "vpc" {
+  source            = "github.com/cds-snc/terraform-modules//vpc?ref=v9.2.5"
+  name              = ${{ values.vpc_name | dump }}
+  billing_tag_value = ${{ values.billing_tag_value| dump }}
```
Comment (patheard) [inline templates/vpc/content/vpc.tf:8]: For the configuration options available I have a few more to request, and I get that this is probably awkward to understand via a comment so more than happy to chat about it tomorrow:

1. High availability?  (true/false)
```hcl
  # true
  availability_zones = 3
  single_nat_gateway = false

  # false
  availability_zones = 1 
  single_nat_gateway = true
```

2. Allow HTTPS requests in (true/false)?  Should users and internet services be able to make requests to something running in the VPC and get a response?
```hcl
  # true
  allow_https_request_in           = true
  allow_https_request_in_response  = true

  # false
  allow_https_request_in           = false
  allow_https_request_in_response  = false
```

3. Allow HTTPS requests out (true/false)?  Is there a service running in the VPC that will be calling out to internet services and expect a response?
```hcl
  # true
  allow_https_request_out          = true
  allow_https_request_out_response = true

  # false
  allow_https_request_out          = false
  allow_https_request_out_response = false
```

File: templates/vpc/content/vpc.tf [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#
+# VPC
+#
+
+module "vpc" {
+  source            = "github.com/cds-snc/terraform-modules//vpc?ref=v9.2.5"
+  name              = ${{ values.vpc_name | dump }}
+  billing_tag_value = ${{ values.billing_tag_value| dump }} 
+}
```
Comment (patheard) [inline templates/vpc/content/vpc.tf:9]: One more config option for Cloud Based Sensor.

1. Is this a staging or production service?  If yes, please provide your AWS target account ID.

```suggestion
  enable_flow_log = false
}

resource "aws_flow_log" "cloud_based_sensor" {
  log_destination      = "arn:aws:s3:::cbs-satellite-${{ values.account_id | dump }}/vpc_flow_logs/"
  log_destination_type = "s3"
  traffic_type         = "ALL"
  vpc_id               = module.vpc.vpc_id
  log_format           = "$${vpc-id} $${version} $${account-id} $${interface-id} $${srcaddr} $${dstaddr} $${srcport} $${dstport} $${protocol} $${packets} $${bytes} $${start} $${end} $${action} $${log-status} $${subnet-id} $${instance-id}"

  tags = {
    CostCentre = ${{ values.billing_tag_value | dump }}
    Terraform  = true
  }
}
```

File: templates/vpc/content/outputs.tf (hunk 1 above)
Comment (sylviamclaughlin) [reply templates/vpc/content/outputs.tf:1]: Sorry about this! it has been ran.

Comment (patheard) [reply templates/vpc/content/outputs.tf:1]: Thank you!

File: templates/vpc/content/vpc.tf (hunk 2 above)
Comment (sylviamclaughlin) [reply templates/vpc/content/vpc.tf]: This has been changed.

File: templates/vpc/content/vpc.tf (hunk 3 above)
Comment (sylviamclaughlin) [reply templates/vpc/content/vpc.tf:28]: I have added a counter to the cloud based sensor to create it if it is a staging or production account. 

Thanks for all the suggestions and feedback Pat!

File: templates/vpc/content/vpc.tf [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  name                              = ${{ values.vpc_name | dump }}
+  billing_tag_value                 = ${{ values.billing_tag_value| dump }} 
+  availability_zones                = ${{ values.availability_zones | dump }}
+  single_nat_gateway                = ${{ values.single_nat_gateway }}
+  allow_https_requests_in           = ${{ values.allow_https_requests_in | dump }}
+  allow_https_requests_in_response  = ${{ values.allow_https_requests_in_response }}
+  allow_https_requests_out          = ${{ values.allow_https_requests_out | dump }}
+  allow_https_requests_out_response = ${{ values.allow_https_requests_out_response }}
+
+  resource "aws_flow_log" "cloud_based_sensor" {
```
Comment (patheard) [inline templates/vpc/content/vpc.tf:16]: Thank you for making all the changes!  Last little tweak is that this `resource` should be outside of the `module` block.

Comment (sylviamclaughlin) [reply templates/vpc/content/vpc.tf]: Sorry Pat! What a silly mistake for me to make - I think I was on autopilot when I made this change. It has been corrected.

**Review by patheard — APPROVED** [review]

**Pull request merged**